components/trello/trello.app.js

```diff
@@ -0,0 +1,240 @@
+const axios = require("axios");
+const events = `addAttachmentToCard
+addChecklistToCard
+addLabelToCard
+addMemberToBoard
+addMemberToCard
+commentCard
+convertToCardFromCheckItem
+copyCard
+createCard
+createCheckItem
+createLabel
+createList
+deleteAttachmentFromCard
+deleteCard
+deleteCheckItem
+deleteComment
+deleteLabel
+emailCard
+moveCardFromBoard
+moveCardToBoard
+moveListFromBoard
+moveListToBoard
+removeChecklistFromCard
+removeLabelFromCard
+removeMemberFromBoard
+removeMemberFromCard
+updateBoard
+updateCard
+updateCheckItem
+updateCheckItemStateOnCard
+updateChecklist
+updateComment
+updateLabel
+updateList`.split("\n");
+
+module.exports = {
+  type: "app",
+  app: "trello",
+  propDefinitions: {
+    cardIds: {
+      // after should be array + assume after apps
+      type: "string[]",
+      label: "Cards",
+      optional: true,
+      // options needs to support standardized opts for pagination
+      async options(opts) {
+        const cards = await this.getCards(opts.boardId);
+        // XXX short hand where value and label are same value
+        return cards.map((card) => {
+          return { label: `${card.name} (${card.id})`, value: card.id };
+        });
+      },
+    },
+    boardId: {
+      // after should be array + assume after apps
+      type: "string",
+      label: "Board",
+      // options needs to support standardized opts for pagination
+      async options(opts) {
+        const boards = await this.getBoards(this.$auth.oauth_uid);
+        // XXX short hand where value and label are same value
+        const activeBoards = boards.filter((board) => board.closed === false);
+        return activeBoards.map((board) => {
+          return { label: `${board.name} (${board.id})`, value: board.id };
+        });
+      },
+      // XXX validate
```

Comment:
Did you still need to do some validation?

Reply:
Not sure. That was written by @psavkar